### PR TITLE
Schema extension support

### PR DESCRIPTION
**Pull request by bbakerman**
This adds support for schema extensions into graphql-java

```
extend schema {
    mutation : SomeNewType
}
```

This also captures the schema extension definition objects and puts them into the runtime schema

This will allow Apollo to do their funky extension trick when stitching together schemas

### CODE REVIEW

File: src/main/antlr/Graphql.g4 [hunk 1]
```diff
@@ -11,7 +11,8 @@ document : definition+;
 definition:
 operationDefinition |
 fragmentDefinition |
-typeSystemDefinition
+typeSystemDefinition |
```
Comment (andimarek) [inline src/main/antlr/Graphql.g4]: why separating it?

File: src/main/java/graphql/schema/GraphQLSchema.java [hunk 2]
```diff
@@ -238,6 +238,38 @@ public GraphQLDirective getDirective(String name) {
...
 
+    /**
+     * This returns the list of directives that have been explicitly put on the
+     * schema object.  Note that {@link {@link #getDirectives()}} will return
+     * directives for all schema elements, whereas this is just for the schema
+     * element itself
+     *
+     * @return a list of directives
+     */
+    public List<GraphQLDirective> getSchemaDirectives() {
```
Comment (andimarek) [inline src/main/java/graphql/schema/GraphQLSchema.java]: maybe add schema directive by name similar to other schema elements

Comment (andimarek) [reply src/main/java/graphql/schema/GraphQLSchema.java]:
````  default Map<String, GraphQLDirective> getDirectivesByName() {
        return directivesByName(getDirectives());
    }```

File: src/main/antlr/Graphql.g4 (hunk 1 above)
Comment (bbakerman) [reply src/main/antlr/Graphql.g4]: exaplained in person

**Pull request merged**